src/librustdoc/html/render/print_item.rs

```diff
@@ -1148,17 +1161,21 @@ fn item_enum(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, e: &clean::
             render_attributes_in_pre(w, it, "");
             write!(
                 w,
-                "{}enum {}{}{}",
+                "{}enum {}{}",
                 it.visibility.print_with_space(it.item_id, cx),
                 it.name.unwrap(),
                 e.generics.print(cx),
-                print_where_clause(&e.generics, cx, 0, true),
             );
+            if !print_where_clause_and_check(w, &e.generics, cx, 0, true) {
+                // If there wasn't a `where` clause, we add a whitespace.
```

Comment:
I think it would be better to make this whitespace emitted by `print_where_clause`. Then the role of `print_where_clause` is simple to explain and self contained. It either (a) prints a newline, a where clause, and another newline, or (b) prints a whitespace.

Reply:
Let me give it a try then!

Reply:
Unfortunately it cannot work for cases like:

```rust
trait A {
    fn foo<T>(a: T)
    where T: Clone;
}

type F<T> where T: Clone = Vec<T>;
trait G<T> where T: Clone = A<T>;
```

If we handle the whitespace directly into `print_where_clause`, it'll need to be handled manually in these cases to "counter" the effect.